src/plugins/expressions/common/execution/execution.ts

```diff
@@ -487,6 +522,12 @@ export class Execution<
           }
         }
 
+        if (hash) {
+          while (this.functionCache!.size >= maxCacheSize) {
+            this.functionCache!.delete(this.functionCache!.keys().next().value);
```

Comment:
nit: `keys()` returns keys based on insertion order, so here it is deleting the oldest one. While this is part of the JS spec it would be useful to leave it as a comment that we're relying on that as cleaning strategy.